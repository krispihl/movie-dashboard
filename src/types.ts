export type Genre = {
    id: number,
    name: string
    popular: boolean;
};

export type Movie = {
    adult: boolean;
    backdrop_path: string;
    genre_ids: number[];
    id: number;
    original_language: string;
    original_title: string;
    overview: string;
    popularity: number;
    poster_path: string;
    release_date: string;
    title: string;
    video: boolean;
    vote_average: number;
    vote_count: number;
};

export type VideoData = {
    id: string;
    iso_639_1: string;
    iso_3166_1: string;
    key: string;
    name: string;
    official: boolean;
    published_at: string;
    site: string;
    size: number;
    type: string;
};

export type MovieWithCustomProps = Movie & {
    rowPosition: number;
};

export type Config = {
    base_url: string;
    secure_base_url: string;
    backdrop_sizes: string[];
    logo_sizes: string[];
    poster_sizes: string[];
    profile_sizes: string[];
    still_sizes: string[];
};

export enum PopularGenres {
    Comedy = 35,
    Drama = 18,
    Action = 28,
    Horror = 27,
    Documentary = 99
};