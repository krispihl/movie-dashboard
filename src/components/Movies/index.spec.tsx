import { Movies } from "./index";
import { render, screen, act } from '@testing-library/react';
import { Movie } from "../../types";

const mockedMovies = [{}, {}] as Movie[];

jest.mock('../Movie', () => ({
    Movie: () => {
        return <div className='mocked-movie' />
    }
}));

jest.mock('../../utils/recursive-fetch', () => ({
    recursiveFetch: () => mockedMovies
}));

jest.mock('../../contexts/movie-details', () => ({
    useMovieDetails: () => jest.fn()
}));

jest.mock('../../contexts/movies', () => ({
    useMovies: () => ({
        setConfig: jest.fn(),
        allMovies: mockedMovies,
        setAllMovies: jest.fn(),
        filteredMovies: mockedMovies,
        setFilteredMovies: jest.fn(),
    })
}));

describe('src/components/Movies', () => {
    it('should render Movies component correctly', async () => {
        global.fetch = jest.fn(() =>
            Promise.resolve({
                json: () => {return { images: {} }}
            })
        ) as jest.Mock;

        let container;

        await act(async () => {
            container = render(<Movies />).container;
        });

        expect(container).toMatchSnapshot();
        expect(screen.getByRole('main')).toBeInTheDocument();
        expect(document.querySelectorAll('.mocked-movie')).toHaveLength(2);
    });
});
