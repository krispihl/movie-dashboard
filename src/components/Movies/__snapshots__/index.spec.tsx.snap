// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/Movies should render Movies component correctly 1`] = `
<div>
  <main
    class="movies"
    id="movies-grid"
  >
    <div
      class="mocked-movie"
    />
    <div
      class="mocked-movie"
    />
  </main>
</div>
`;
